presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -363,13 +363,40 @@ public void testPushdownArraySubscripts()
         assertPushdownSubfields("SELECT id, b[1] FROM test_pushdown_array_subscripts CROSS JOIN UNNEST(a) as t(unused)", "test_pushdown_array_subscripts",
                 ImmutableMap.of("b", toSubfields("b[1]")));
 
+        // Aggregation
+        assertPushdownSubfields("SELECT arbitrary(y[1]).a FROM test_pushdown_array_subscripts GROUP BY id", "test_pushdown_array_subscripts",
+                ImmutableMap.of("y", toSubfields("y[1].a")));
+
+        assertPushdownSubfields("SELECT arbitrary(y[1]).d.d1 FROM test_pushdown_array_subscripts GROUP BY id", "test_pushdown_array_subscripts",
+                ImmutableMap.of("y", toSubfields("y[1].d.d1")));
+
+        assertPushdownSubfields("SELECT arbitrary(y[2].d).d1 FROM test_pushdown_array_subscripts GROUP BY id", "test_pushdown_array_subscripts",
+                ImmutableMap.of("y", toSubfields("y[2].d.d1")));
+
+        assertPushdownSubfields("SELECT arbitrary(y[3].d.d1) FROM test_pushdown_array_subscripts GROUP BY id", "test_pushdown_array_subscripts",
+                ImmutableMap.of("y", toSubfields("y[3].d.d1")));
+
+        assertPushdownSubfields("SELECT arbitrary(z[1][2]).e.e1 FROM test_pushdown_array_subscripts GROUP BY id", "test_pushdown_array_subscripts",
+                ImmutableMap.of("z", toSubfields("z[1][2].e.e1")));
+
+        assertPushdownSubfields("SELECT arbitrary(z[2][3].e).e2 FROM test_pushdown_array_subscripts GROUP BY id", "test_pushdown_array_subscripts",
+                ImmutableMap.of("z", toSubfields("z[2][3].e.e2")));
+
         // No subfield pruning
         assertPushdownSubfields("SELECT array_sort(a)[1] FROM test_pushdown_array_subscripts", "test_pushdown_array_subscripts",
                 ImmutableMap.of());
 
         assertPushdownSubfields("SELECT id FROM test_pushdown_array_subscripts CROSS JOIN UNNEST(a) as t(index) WHERE a[1] > 10 AND cardinality(b[index]) = 2", "test_pushdown_array_subscripts",
                 ImmutableMap.of());
 
+        // No pass through subfield pruning
+        assertPushdownSubfields("SELECT id, min(y[1]).a FROM test_pushdown_array_subscripts GROUP BY 1", "test_pushdown_array_subscripts",
```

Comment:
can this go into assertPushdownSubscripts?

Reply:
Done